src/main/kotlin/deadline/application/DeadlineChanger.kt

```diff
@@ -0,0 +1,34 @@
+package deadline.application
+
+import deadline.domain.Deadline
+import deadline.domain.DeadlineRepository
+import deadline.domain.Time
+import deadline.domain.exceptions.AlreadyConfiguredDeadline
+import deadline.domain.exceptions.NonExistentDeadlineException
+import shared.domain.Identifier
+import shared.domain.exceptions.InvalidUUIDException
+
+class DeadlineChanger(private val repository: DeadlineRepository) {
+
+    /***
+     * Changes the current deadline date to the new deadline values
+     * @throws InvalidUUIDException if the id from the primitives isn't valid
+     * @throws NonExistentDeadlineException if the deadline id from the primitives doesn't exist on the repository
+     * @throws AlreadyConfiguredDeadline if the values of the current deadline are the same as the new one
+     */
+    fun change(newDeadline: DeadlinePrimitives) = change(Identifier(newDeadline.noteIdentifier), Time(newDeadline.time))
+
+    private fun change(noteId: Identifier, time: Time): Deadline {
+        val currentDeadline = repository.search(noteId) ?: throw NonExistentDeadlineException()
```

Comment:
Shouldn't we distinguish between the scenario where the note doesn't exist and the one where the note exists but there's no deadline configured for it?

Reply:
Here we're updating a deadline, so we're assuming that the note already exists and we only check if the relationship this note -> deadline is already created, and if not, throw an exception.

We could check if the note exists too, and I am assuming that is what you're telling me here

Reply:
Yes, I meant to check if the note exists too.